frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1055,6 +1064,44 @@ pub mod pallet {
 			<QueuedSolution<T>>::put(solution);
 			Ok(())
 		}
+
+		#[pallet::weight(100)]
+		pub fn challenge_solution(origin: OriginFor<T>, index: u32) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			// TODO: Create custom error
+			ensure!(
+				T::Currency::can_slash(&who, T::MinimumSlashableAmount::get()),
+				<Error<T>>::CallNotAllowed
+			);
+			// TODO: Add an `else` for a None result
+			if let Some(submission) = SignedSubmissionsMap::<T>::get(index) {
+				match Self::feasibility_check(
+					submission.raw_solution.clone(),
+					ElectionCompute::Signed,
+				) {
+					Ok(solution) => {
+						let _ = T::Currency::slash(&who, T::MinimumSlashableAmount::get());
+						CheckedSolutions::<T>::insert(submission, solution);
```

Comment:
Here you must insert the good one into the `QueuedSolution`, and you must clean it from `SignedSubmissions`, because it has been already early-verified. Not sure about what to do with the reward in this case though, any thoughts @emostov? 

Alternative is to just not sure this solution into `QueuedSolution`, and not treat it as "already-verified" at all. This is certainly simpler to code, and we can find a way to reuse this verified solution later.